rankboost.py

```diff
@@ -437,6 +443,88 @@ def prepare_dataframe(mhc, predfile, expfile, pepfile, efilt_threshold, boost):
     return all_data, stats, boost
 
 
+def collapse_iars(stats):
+    """
+    Collapse duplicated IARs.
+
+    :param pandas.core.frame.DataFrame stats: The stats dataframe generated from parsing the input
+           calls
+    :return: A collapsed dataframe where overlapping IARS are merged into a single call
+    :rtype: pandas.core.frame.DataFrame
+
+    >>> df = pandas.DataFrame({'IAR_sequence': ['ABCDE', 'FGHI', \
+                                            'ABCD',          '12345', \
+                                            'ABCDE', 'FGHIJ', \
+                                            'ABC',                    '7891011'], \
+                           'mutations': ['A,B', 'G', 'A', '1', 'A,C', 'G,H', 'A,D', '8']}, \
+                           index=['ne1', 'ne2', 'ne3', 'ne4', 'ne5', 'ne6', 'ne7', 'ne8'])
+    >>> tst = collapse_iars(df)
+    >>> tst.index
+    Index([u'ne1', u'ne2', u'ne4', u'ne8'], dtype='object')
+    >>> tst.loc['ne1', 'mutations']
+    'A,B,C,D'
+    >>> tst.loc['ne2', 'mutations']
+    'G,H'
+    >>> tst.loc['ne4', 'mutations']
+    '1'
+    >>> tst.loc['ne8', 'mutations']
+    '8'
+    >>> tst.loc['ne1', 'epitope_name(s)']
+    'ne1,ne3,ne5,ne7'
+    >>> tst.loc['ne2', 'epitope_name(s)']
+    'ne2,ne6'
+    >>> tst.loc['ne4', 'epitope_name(s)']
+    'ne4'
+    >>> tst.loc['ne8', 'epitope_name(s)']
+    'ne8'
+    """
+    # Add new column for identical neoepitopes and for tagging rows to drop
+    stats['epitope_name(s)'] = stats.index
+    stats['collapsed'] = False
+    for epitope1 in stats.index:
+        # Get the IAR to compare against other IARs
+        iar_to_check = stats.loc[epitope1]['IAR_sequence']
+        # Create sets for extending names and mutations
+        joined_names = list()
+        joined_names.append(stats.loc[epitope1, 'epitope_name(s)'])
+        joined_mutations = stats.loc[epitope1, 'mutations'].split(',')
+        first = True
+        for epitope2 in stats[epitope1:].index:
+            # First epitope1 == epitope2, skip it
+            if first:
+                first = False
+                continue
+            # If the IAR is the same or is a part of a different IAR while not being the same
+            # neoepitope, it has to be collapsed
+            if ((iar_to_check in stats.loc[epitope2]['IAR_sequence']
+                    or stats.loc[epitope2]['IAR_sequence'] in iar_to_check)
+                    and not stats.loc[epitope2, 'collapsed']):
+                # Add the neoepitope name to the extended group
+                joined_names.append(stats.loc[epitope2, 'epitope_name(s)'])
+                # Add the mutation to the extended group
+                mutations = stats.loc[epitope2, 'mutations'].split(',')
+                for mutation in mutations:
```

Comment:
you could probably make `joined_mutaitons` a set and just use `joined_mutations.update(mutations)`

Reply:
instead of `for: if`